src/Utilities/mockoptimizer.jl

```diff
@@ -286,21 +286,39 @@ end
 mock_optimize!(mock::MockOptimizer, primdual...) = mock_optimize!(mock, MOI.Success, primdual...)
 
 # Sets variable primal to varprim
-function mock_primal!(mock::MockOptimizer, primstatus::MOI.ResultStatusCode, varprim::Vector, dual...)
+function mock_primal!(mock::MockOptimizer, primstatus::MOI.ResultStatusCode, varprimdual...)
     MOI.set!(mock, MOI.PrimalStatus(), primstatus)
-    MOI.set!(mock, MOI.VariablePrimal(), MOI.get(mock, MOI.ListOfVariableIndices()), varprim)
-    mock_dual!(mock, dual...)
-end
-function mock_primal!(mock::MockOptimizer, varprim::Vector, dual...)
-    # Feasible primal solution
-    mock_primal!(mock, MOI.FeasiblePoint, varprim, dual...)
+    mock_varprimal!(mock, varprimdual...)
 end
+# Shortcut for feasible primal solution
+mock_primal!(mock::MockOptimizer, varprim::Vector, dual...) = mock_primal!(mock, MOI.FeasiblePoint, varprim, dual...)
 function mock_primal!(mock::MockOptimizer, conduals::Pair...)
     # No primal solution
     mock.hasprimal = false
+    mock_varprimal!(mock, conduals...)
+end
+
+# Sets variable primal to varprim
+function mock_varprimal!(mock::MockOptimizer, varprim::Vector, dual...)
+    MOI.set!(mock, MOI.VariablePrimal(), MOI.get(mock, MOI.ListOfVariableIndices()), varprim)
+    mock_dual!(mock, dual...)
```

Comment:
Why does `mock_varprimal!` call `mock_dual!` which calls `mock_condual!?

Reply:
The reader shouldn't need to go through 6 levels of recursion to understand all of the properties that are being set.